src/tracer.rs

```diff
@@ -22,6 +26,45 @@ impl Tracer {
     }
 }
 
+#[macro_export]
+/// Provides dependencies for the Tracer.
+/// Designed to be imported into a user's module at the root level.
+macro_rules! tracer_dependencies {
```

Comment:
Is there a reason to make this separate from the `tracer_bootstrap!` macro? It seems to me that it could be within the same macro, and it would make it easier for users.

Reply:
I asked on IRC because I wasn't sure about this, but it seems like if the user has already done something like
```rust
use std::sync::Mutex;
```
in their application, Cargo will error on the double-definition of `Mutex`. Is there a cleaner way?

Reply:
I actually think that any macro might be completely unnecessary. The point is to create the static ring buffer, right? That can be done inside the `wasm_trace` runtime crate, and then users would only need to add the `extern crate wasm_trace;` line.

Reply:
Ah, okay! Do you mean inside `lib.rs`? I don't have a great mental model for how runtime linking works.

Reply:
`extern crate foo;` does static linking, so unless things have changed since I last looked, then everything is statically linked right now.

> Do you mean inside lib.rs?

That would work, or anywhere else in your crate. The key thing is that it needs to be transitively used by some function that gets exported so that `wasm-gc`/`rustc` don't think it is dead code and remove it.